databricks/koalas/series.py

```diff
@@ -4215,6 +4215,11 @@ def __len__(self):
         return len(self.to_dataframe())
 
     def __getitem__(self, key):
+        if isinstance(key, Series) and isinstance(key.spark_type, BooleanType):
+            self._kdf["__temp_col__"] = key
+            sdf = self._kdf._sdf.filter(F.col("__temp_col__")).drop("__temp_col__")
+            return _col(DataFrame(_InternalFrame(sdf=sdf, index_map=self._internal.index_map)))
```

Comment:
hm, actually it should set column index. In `where` too .. I see some related problems .. let me fix it separately.